diff --git a/apps/main.c b/apps/main.c
index 1e8e872296..f3b5555def 100644
--- a/apps/main.c
+++ b/apps/main.c
@@ -160,16 +160,15 @@ static void init(void);
  * be INIT_ATTR. These functions must not be called after the final call
  * to root_menu() at the end of main()
  * see definition of INIT_ATTR in config.h */
-#ifdef HAVE_ARGV_MAIN
-int main(int argc, char *argv[]) INIT_ATTR MAIN_NORETURN_ATTR ;
-int main(int argc, char *argv[])
-{
-    sys_handle_argv(argc, argv);
-#else
-int main(void) INIT_ATTR MAIN_NORETURN_ATTR;
-int main(void)
+
+__attribute__((constructor))
+int app_start()
 {
-#endif
+    putenv("SDL_VIDEODRIVER=fbcon");
+    putenv("SDL_FBDEV=/dev/fb");
+    putenv("SDL_AUDIODRIVER=dsp");
+    putenv("SDL_PATH_DSP=/dev/snd/dsp");
+
     CHART(">init");
     init();
     CHART("<init");
