diff --git a/uisimulator/buttonmap/sansa-fuzeplus.c b/uisimulator/buttonmap/sansa-fuzeplus.c
index ef50009a35..0a181247e1 100644
--- a/uisimulator/buttonmap/sansa-fuzeplus.c
+++ b/uisimulator/buttonmap/sansa-fuzeplus.c
@@ -21,6 +21,7 @@
 
 
 #include <SDL.h>
+#include <unistd.h>
 #include "button.h"
 #include "buttonmap.h"
 
@@ -29,48 +30,40 @@ int key_to_button(int keyboard_button)
     int new_btn = BUTTON_NONE;
     switch (keyboard_button)
     {
-        case SDLK_KP4:
-        case SDLK_LEFT:
+        case 5:
             new_btn = BUTTON_LEFT;
             break;
-        case SDLK_KP6:
-        case SDLK_RIGHT:
+        case 7:
             new_btn = BUTTON_RIGHT;
             break;
-        case SDLK_KP8:
-        case SDLK_UP:
+        case 3:
             new_btn = BUTTON_UP;
             break;
-        case SDLK_KP2:
-        case SDLK_DOWN:
+        case 9:
             new_btn = BUTTON_DOWN;
             break;
-        case SDLK_PAGEUP:
-        case SDLK_KP9:
+        case 28:
             new_btn = BUTTON_PLAYPAUSE;
             break;
-        case SDLK_KP7:
+        case 14:
             new_btn = BUTTON_BACK;
             break;
-        case SDLK_KP5:
-        case SDLK_SPACE:
-        case SDLK_KP_ENTER:
-        case SDLK_RETURN:
+        case 6:
             new_btn = BUTTON_SELECT;
             break;
-        case SDLK_KP_PLUS:
+        case 69:
             new_btn = BUTTON_VOL_UP;
             break;
-        case SDLK_KP_MINUS:
+        case 102:
             new_btn = BUTTON_VOL_DOWN;
             break;
-        case SDLK_HOME:
+        case 223:
             new_btn = BUTTON_POWER;
             break;
-        case SDLK_KP1:
+        case 8:
             new_btn = BUTTON_BOTTOMLEFT;
             break;
-        case SDLK_KP3:
+        case 10:
             new_btn = BUTTON_BOTTOMRIGHT;
             break;
     }
@@ -78,17 +71,17 @@ int key_to_button(int keyboard_button)
 }
 
 struct button_map bm[] = {
-    { SDLK_KP7,        69, 401, 39, "Back" },
-    { SDLK_KP8,       161, 404, 34, "Up" },
-    { SDLK_KP9,       258, 400, 43, "Play/Pause" },
-    { SDLK_KP4,        69, 477, 36, "Left" },
-    { SDLK_KP5,       161, 476, 31, "Select" },
-    { SDLK_KP6,       222, 474, 41, "Right" },
-    { SDLK_KP1,        82, 535, 34, "Bottom-Left" },
-    { SDLK_KP2,       162, 532, 33, "Down" },
-    { SDLK_KP3,       234, 535, 42, "Bottom-Right" },
-    { SDLK_KP_PLUS,     1, 128, 29, "Vol+" },
-    { SDLK_KP_MINUS,    5, 187, 30, "Vol-" },
-    { SDLK_HOME,      170,   6, 50, "Power" },
+    { 14,        69, 401, 39, "Back" },
+    { 3,       161, 404, 34, "Up" },
+    { 28,       258, 400, 43, "Play/Pause" },
+    { 5,        69, 477, 36, "Left" },
+    { 6,       161, 476, 31, "Select" },
+    { 7,       222, 474, 41, "Right" },
+    { 8,        82, 535, 34, "Bottom-Left" },
+    { 9,       162, 532, 33, "Down" },
+    { 10,       234, 535, 42, "Bottom-Right" },
+    { 69,     1, 128, 29, "Vol+" },
+    { 102,    5, 187, 30, "Vol-" },
+    { 223,      170,   6, 50, "Power" },
     { 0, 0, 0, 0, "None" }
 };
